reframe/core/pipeline.py

```diff
@@ -1085,14 +1098,40 @@ def run(self):
         This call is non-blocking.
         It simply submits the job associated with this test and returns.
         """
+
         if not self.current_system or not self._current_partition:
             raise PipelineError('no system or system partition is set')
 
+        if self.container_platform:
+            try:
+                cp_name = self.container_platform.__class__.__name__
```

Comment:
Rewrite this as follows:

```suggestion
                cp_name = type(self.container_platform).__name__
```